examples/text_classification_with_scriptable_tokenzier/train_model.py

```diff
@@ -0,0 +1,177 @@
+optim = AdamW(model.parameters(), lr=LEARNING_RATE)
+criteria = nn.CrossEntropyLoss()
+
+
+def train_step(input_tensor, target):
+    """
+    Performs a training step
+
+    Args:
+        input: Input data for step
+        target: Target for step
+    """
+    output = model(input_tensor)
+    loss = criteria(output, target)
+    optim.zero_grad()
+    loss.backward()
+    optim.step()
+
+
+def eval_step(input_tensor, target):
+    """
+    Performs a evaluation step
+
+    Args:
+        input: Input data for step
+        target: Target for step
+
+    Return:
+        (Tuple): Loss and accuracy
+    """
+    output = model(input_tensor)
+    loss = criteria(output, target).item()
+    return float(loss), (output.argmax(1) == target).type(torch.float).sum().item()
+
+
+def evaluate():
+    """
+    Performs a evaluation
+
+    Return:
+        (Tuple): Loss and accuracy
+    """
+    model.eval()
+    total_loss = 0
+    correct_predictions = 0
+    total_predictions = 0
+    counter = 0
+    with torch.no_grad():
+        for batch in dev_dataloader:
+            input_tensor = F.to_tensor(batch["token_ids"], padding_value=PADDING_IDX).to(DEVICE)
+            target = torch.tensor(batch["target"]).to(DEVICE)
+            loss, predictions = eval_step(input_tensor, target)
+            total_loss += loss
+            correct_predictions += predictions
+            total_predictions += len(target)
+            counter += 1
+
+    return total_loss / counter, correct_predictions / total_predictions
+
+
+def train():
+    """
+    Performs a training
+    """
+    num_epochs = 1
+
+    for epoch in range(num_epochs):
+        for batch in train_dataloader:
+            input_tensor = F.to_tensor(batch["token_ids"], padding_value=PADDING_IDX).to(DEVICE)
+            target = torch.tensor(batch["target"]).to(DEVICE)
+            train_step(input_tensor, target)
+
+        loss, accuracy = evaluate()
+        print(f"Epoch = [{epoch}], loss = [{loss}], accuracy = [{accuracy}]")
+
+
+train()
+
+
+class TokenizerModelAdapter(nn.Module):
+    """
+    TokenizerModelAdapter moves input onto device and adds batch dimension
+    """
+    def __init__(self, padding_idx):
+        super().__init__()
+        self._padding_idx = padding_idx
+        self._dummy_param = nn.Parameter(torch.empty(0))
+
+    def forward(self, tokens: Any) -> torch.Tensor:
+        """
+        Moves input onto device and adds batch dimension.
+
+        Args:
+            x (Any): tokenizer output
+
+        Returns:
+            (Tensor): On device text tensor with batch dimension
+        """
+        tokens = F.to_tensor(tokens, padding_value=self._padding_idx).to(self._dummy_param.device)
+        if len(tokens.shape) == 2:
+            return tokens
+        else:
+            return tokens.unsqueeze(0)
+
+combi_model = T.Sequential(
+    text_transform,
+    TokenizerModelAdapter(PADDING_IDX),
+    model,
+)
+
+combi_model.eval()
+
+combi_model.to("cpu")
```

Comment:
Is the training happening on GPU if available but then model scripted on CPU?

Reply:
Yes, for scripting the model should be moved to cpu. Will put a comment here.